### DISCUSSION TITLE
cloning and compiling PR fork

### DISCUSSION THREAD

**Discussion by runlow** [nesbox/TIC-80#1676] [Q&A] [answered]
If I made a fork with only one line different from upstream - I'd imagine that you'd have to clone my repository in PR and compile it. It takes a lot of time. Is there some kind of shortcut around this?

I'll try to word it differently - if you have a clone of https://github.com/nesbox/TIC-80 on your HDD and you change a line - recompiling it will take just a moment because there already being `*.o` files around. If someone makes a PR with the exact same change you'd have to clone their repository and compile it which could take up to 10 minutes depending on the machine.

In other words - is there some way of telling `git` *"hey there's this fork that's only a little bit different, bring in what's different for me to test something and I'll set it back the way it was"*

Am I making sense?

This is not specifically TIC-80 related so apologies. Feel free to link me if you don't feel like explaining it.

If there's no way there's no way (or not practical), was just wondering.

**Comment by joshgoebel** [comment, marked as answer]
I think your looking for 'git remote'. Just add other forks as remotes to a single directory.

**Reply by runlow** [reply]
Interesting! Knew that you could have multiple remotes but wasn't sure what it's for. Thought it's maybe for backup if the upstream goes down.

Didn't know you could use it for forks. I'll see if I can get it to work and will mark this is the answer.

**Reply by runlow** [reply]
This seems to work, thank you
```
git remote add joshgoebel https://github.com/joshgoebel/TIC-80.git
git pull joshgoebel
git checkout foo-branch-name
make -j4 -C build
```

**Reply by joshgoebel** [reply]
I use `git -fetch all` a lot vs pull... but yeah that's the idea. :-)
